amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueServiceImpl.java

```diff
@@ -0,0 +1,295 @@
+ * Logic of queue.
+ */
+@Slf4j
+public class QueueServiceImpl implements QueueService {
+    private ExchangeContainer exchangeContainer;
+    private QueueContainer queueContainer;
+
+    public QueueServiceImpl(ExchangeContainer exchangeContainer,
+                            QueueContainer queueContainer) {
+        this.exchangeContainer = exchangeContainer;
+        this.queueContainer = queueContainer;
+    }
+
+    @Override
+    public void queueDeclare(AmqpChannel channel, AMQShortString queue, boolean passive, boolean durable,
+                             boolean exclusive,
+                             boolean autoDelete, boolean nowait, FieldTable arguments) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    "RECV[{}] QueueDeclare[ queue: {}, passive: {}, durable:{}, "
+                            + "exclusive:{}, autoDelete:{}, nowait:{}, arguments:{} ]",
+                    channelId, queue, passive, durable, exclusive, autoDelete, nowait, arguments);
+        }
+        if ((queue == null) || (queue.length() == 0)) {
+            queue = AMQShortString.createAMQShortString("tmp_" + UUID.randomUUID());
+        }
+        AMQShortString finalQueue = queue;
+        CompletableFuture<AmqpQueue> amqpQueueCompletableFuture =
+                queueContainer.asyncGetQueue(connection.getNamespaceName(), finalQueue.toString(), !passive);
+        amqpQueueCompletableFuture.whenComplete((amqpQueue, throwable) -> {
+            if (throwable != null) {
+                log.error("Get Topic error:{}", throwable.getMessage());
+                channel.closeChannel(INTERNAL_ERROR, "Get Topic error: " + throwable.getMessage());
+            } else {
+                if (null == amqpQueue) {
+                    channel.closeChannel(ErrorCodes.NOT_FOUND, "No such queue: " + finalQueue);
+                } else {
+                    channel.checkExclusiveQueue(amqpQueue);
+                    channel.setDefaultQueue(amqpQueue);
+
+                    MethodRegistry methodRegistry = connection.getMethodRegistry();
+                    QueueDeclareOkBody responseBody = methodRegistry.createQueueDeclareOkBody(finalQueue, 0, 0);
+                    connection.writeFrame(responseBody.generateFrame(channelId));
+                }
+            }
+        });
+    }
+
+    @Override
+    public void queueDelete(AmqpChannel channel, AMQShortString queue, boolean ifUnused,
+                            boolean ifEmpty, boolean nowait) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug("RECV[{}] QueueDelete[ queue: {}, ifUnused:{}, ifEmpty:{}, nowait:{} ]", channelId, queue,
+                    ifUnused, ifEmpty, nowait);
+        }
+        if ((queue == null) || (queue.length() == 0)) {
```

Comment:
The default queue is the queue which declared in the last of the channel. If user call delete queue without queue name, it will delete the default queue.